backend/fms_core/template_importer/row_handlers/library_preparation/library_preparation.py

```diff
@@ -0,0 +1,36 @@
+from fms_core.template_importer.row_handlers._generic import GenericRowHandler
+
+from fms_core.services.library import prepare_library
+from fms_core.services.library import get_library_type
+from fms_core.services.platform import get_platform
+
+
+class LibraryPreparationRowHandler(GenericRowHandler):
+    def __init__(self):
+        super().__init__()
+
+    def process_row(self, **kwargs):
+        if len(kwargs['sample_rows_info']) < 1:
+            self.errors['samples'] = f"No samples are associated to this library"
+
+        return super(self.__class__, self).process_row(**kwargs)
+
+    def process_row_inner(self, library_type, library_size, library_date, platform, comment,
+                          library_rows_info, protocol, process_properties):
+
+        library_type_obj, self.errors['experiment_type'], self.warnings['experiment_type'] = \
+            get_library_type(library_type['name'])
+
+        platform_obj, self.errors['platform'], self.warnings['platform'] = get_platform(platform['name'])
+
+        if library_type_obj and platform_obj:
+            _, self.errors['library'], self.warnings['library'] = prepare_library(library_type_obj,
+                                                                                  library_size,
+                                                                                  platform_obj,
+                                                                                  library_date,
+                                                                                  library_rows_info,
+                                                                                  protocol,
+                                                                                  process_properties,
+                                                                                  comment)
+        else:
+            self.errors['library'] = f"Errors prevent the creation of the library."
```

Comment:
We are at the level of the process here, you should mention that the "Library Preparation" failed. Maybe assign the error to something else than library. maybe library_preparation or simply preparation.